OpenRA.Mods.RA/Activities/Teleport.cs

```diff
@@ -53,16 +55,71 @@ public override Activity Tick(Actor self)
 				if (condition.PreventsTeleport(self))
 					return NextActivity;
 
-			var bestCell = ChooseBestDestinationCell(self, destination);
-			if (bestCell == null)
-				return NextActivity;
+			if (!force)
+			{
+				var bestCell = ChooseBestDestinationCell(self, destination);
+				if (bestCell == null)
+					return NextActivity;
 
-			destination = bestCell.Value;
+				destination = bestCell.Value;
+			}
 
 			Game.Sound.Play(sound, self.CenterPosition);
 			Game.Sound.Play(sound, self.World.Map.CenterOfCell(destination));
 
-			self.Trait<IPositionable>().SetPosition(self, destination);
+			var pos = self.Trait<IPositionable>();
+			if (pos.CanEnterCell(destination))
+				force = false;
+			pos.SetPosition(self, destination);
+			if (force)
+			{
+				var subcell = pos.OccupiedCells().First(p => p.First == destination).Second;
+				var health = self.TraitOrDefault<Health>();
+				var givenDamage = health != null ? (health.HP + health.MaxHP) / 2 : 0;
+
+				foreach (var a in self.World.ActorMap.GetActorsAt(destination))
```

Comment:
Name this `actor` please.
There is no need to use a single-character variable name.